File: backend/app/schemas/listing_document.py
"""Schemas for downloadable listing documents."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ListingDocumentRead(BaseModel):
    id: int
    listing_id: int
    title: str
    original_filename: str
    content_type: str
    file_size: int
    is_public: bool
    download_url: str
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class PublicListingDocumentRead(BaseModel):
    id: int
    title: str
    original_filename: str
    content_type: str
    file_size: int
    download_url: str


class ListingDocumentUpdate(BaseModel):
    title: str | None = Field(None, min_length=1, max_length=160)
    is_public: bool | None = None

File: frontend/src/app/models/listing-document.ts
// Downloadable PDF metadata used by admin and public listing pages.

export interface PublicListingDocument {
  id: number;
  title: string;
  original_filename: string;
  content_type: string;
  file_size: number;
  download_url: string;
}

export interface ListingDocument extends PublicListingDocument {
  listing_id: number;
  is_public: boolean;
  created_at: string;
  updated_at: string;
}

export interface ListingDocumentUpdate {
  title?: string;
  is_public?: boolean;
}
